client/webpack.dev.js

```diff
@@ -3,13 +3,25 @@ const { merge } = require('webpack-merge');
 const common = require('./webpack.common');
 
 const DEV_SERVER_PORT = 8080;
+const DEFAULT_LOCALHOST_HOST = 'localhost:5000';
 
 module.exports = merge(common, {
   mode: 'development',
   output: {
     filename: '[name].js',
-    publicPath: `//localhost:${DEV_SERVER_PORT}/webpack/`,
     pathinfo: false,
+
+    /**
+     * If the host name of the app (e.g., `localhost:5000`) is different from
+     * that of webpack-dev-server's (e.g., `localhost:8080`), worker scripts
+     * and assets packed by webpack will be hosted under webpack-dev-server's
+     * host name. Accessing these resources from the app's host name will
+     * trigger `SecurityError` in-browser due to the different origins. Forcing
+     * webpack-dev-server's `publicPath` to the app's host name bypasses this,
+     * but may not work if the app is hosted on multiple different domains,
+     * e.g., on both `localhost` and ngrok.
+     */
+    publicPath: `//${DEFAULT_LOCALHOST_HOST}/webpack/`,
```

Comment:
I believe this is to override CORS right? In that case, I think you can instead add rack cors to our backend and add the config to `development.rb`
https://www.stackhawk.com/blog/rails-cors-guide/

Reply:
I have tried using `rack-cors` and allow `localhost:*` to access `/webpack/*`, and yet I still get the `SecurityError` and can't execute the worker from `localhost:8080` in `localhost:5000`.

Seems like what I mentioned yesterday was correct. While the nature of the error is CORS, but it's slightly different because it's _execution_, not just access like typical subpages or assets. It was discussed [here](https://stackoverflow.com/questions/25458104/can-should-html5-web-workers-use-cors-for-cross-origin).

It shouldn't be a problem in production because everything will be served under the same origin. I guess for development, we'll just have to manually change `DEFAULT_LOCALHOST_HOST` in `webpack.dev.js` until we fully migrate to React and our client is self-hosting.